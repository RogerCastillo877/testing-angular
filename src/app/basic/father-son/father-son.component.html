<h4>Son</h4>
<code>{{ client | json }}</code>
<br><br>

<button *ngIf="client" class="btn btn-primary" (click)="onChange(3)" data-test="btn-change-id">Cambiar Id</button>

<button *ngIf="client" class="btn btn-danger" (click)="onDelete()" data-test="btn-delete">Delete Client</button>