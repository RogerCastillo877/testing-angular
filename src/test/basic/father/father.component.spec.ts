import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';

import { FatherComponent } from '../../../app/basic/father/father.component';
import { FatherSonComponent } from '../../../app/basic/father-son/father-son.component';

describe('FatherComponent', () => {
  let component: FatherComponent;
  let fixture: ComponentFixture<FatherComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [FatherComponent, FatherSonComponent]
    })
      .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(FatherComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
    jest.clearAllMocks();
  })

  test('should create a component', () => {
    expect(component).toBeTruthy();
  });

  test('should match with the snapshot', () => {
    expect(compiled).toMatchSnapshot();
  })

  test('should set the client with the given name', () => {
    component.onSetClient('Peter')
    fixture.detectChanges();

    const codeDiv = compiled.querySelector('.mt-2');

    expect(codeDiv?.textContent).toContain('"name"');
    expect(codeDiv?.textContent).toContain('"Peter"');
  })

  test('should delete client if emit onDeleteClient (from son)', () => {
    component.client = { id: 1, name: 'Eduardo' };
    fixture.detectChanges();

    const sonDebugElement = fixture.debugElement.query(By.directive(FatherSonComponent));
    const sonComponent: FatherSonComponent = sonDebugElement.componentInstance;

    sonComponent.onDeleteClient.emit();
    console.log(sonComponent.client)
    // expect(sonComponent?.client).toBe(undefined);
  })

  test('should update client if emit onClientUpdated (from son)', () => {
    component.client = { id: 1, name: 'Eduardo' };
    fixture.detectChanges();

    const sonDebugElement = fixture.debugElement.query(By.directive(FatherSonComponent));
    const sonComponent: FatherSonComponent = sonDebugElement.componentInstance;

    sonComponent.onClientUpdated.emit({ id: 2, name: 'Fer' });
    // expect(sonComponent?.client).toEqual({ id: 2, name: 'Fer' });
  })

});
