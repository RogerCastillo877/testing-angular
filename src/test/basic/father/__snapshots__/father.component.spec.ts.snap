// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FatherComponent should match with the snapshot 1`] = `
<div
  id="root1"
>
  <h2>
    Father
  </h2>
  <button
    class="btn btn-primary"
  >
    Charge Client
  </button>
  <br />
  <br />
  <code
    class="mt-2"
  >
    
  </code>
  <hr />
  <app-father-son>
    <h4>
      Son
    </h4>
    <code>
      
    </code>
    <br />
    <br />
    
    
  </app-father-son>
</div>
`;
